analyses/malawi/dryspells_trigger/07_ecmwf_monthly_skill_general.md

```diff
@@ -41,164 +37,372 @@ reload(processing)
+    leadtimes=df.leadtime.sort_values().unique()
+    df_diff=pd.DataFrame(leadtimes,columns=["leadtime"]).set_index('leadtime')
+    for ilt, leadtime in enumerate(leadtimes):
+        df_lt=df[df.leadtime==leadtime]
+        med,me,std,ci_hi,ci_lo=calc_diff_stats(df_lt[obs_col],df_lt[for_col])
+        df_diff.loc[leadtime,"all_med"]=med
+        df_diff.loc[leadtime,"all_mean"]=me
+        df_diff.loc[leadtime,"all_ci_lo"]=ci_lo
+        df_diff.loc[leadtime,"all_ci_hi"]=ci_hi
+        if threshold_list:
+            for thresh in threshold_list:
+                df_lt_thresh=df_lt[df_lt[obs_col]<=thresh]
+                med,me,std,ci_hi,ci_lo = calc_diff_stats(df_lt_thresh[obs_col],df_lt_thresh[for_col])
+                df_diff.loc[leadtime,f"thresh_{thresh}_med"]=med
+        if month_list:
+            for m in month_list:
+                month_str="".join([calendar.month_abbr[i].lower() for i in m])
+                df_lt_m=df_lt[df_lt.time.dt.month.isin(m)]
+                med,me,std,ci_hi,ci_lo = calc_diff_stats(df_lt_m[obs_col],df_lt_m[for_col])
+                df_diff.loc[leadtime,f"{month_str}_med"]=med
+                df_diff.loc[leadtime,f"{month_str}_ci_lo"]=ci_lo
+                df_diff.loc[leadtime,f"{month_str}_ci_hi"]=ci_hi
+                
+    return df_diff
+```
+
+```python
+df_diff=compute_diff_cats(df_forobs,"precip_obs","precip_for",threshold_list=[210,180,170],month_list=[[m] for m in range(1,13)])
+```
+
+```python
+fig, ax = plt.subplots()
+ax.plot(df_diff.index,df_diff["all_mean"],label="mean")
+ax.fill_between(df_diff.index, df_diff.all_ci_lo, df_diff.all_ci_hi,alpha=0.2,label="confidence interval")
+ax.plot(df_diff.index, df_diff["all_med"],label="median")
+ax.set_xlabel("leadtime [months]")
+ax.set_ylabel("median bias")
+ax.grid()
+ax.legend(bbox_to_anchor=(1.05, 1));
+```
+
+When looking per month though, we can see that the average bias clearly differs per month. The months around the rainy season have the highest bias (nov, dec, jan, feb). This can either be because values are generally larger, or because the forecast has less skill. 
+
+```python
+px.line(df_diff.reset_index(), 
+        x='leadtime', 
+        y=[c for c in df_diff.columns if "thresh" not in c and "med" in c],
+        labels={"value": "median bias"}
+       )
+```
+
+We can also look at the bias for different thresholds. Here we see that there is not much difference. This is however partly caused by the fact that most months have very low values. If you would separate by month and threshold, you would probably get different patterns
+
+```python
+px.line(df_diff.reset_index(), 
+        x='leadtime', 
+        y=[c for c in df_diff.columns if "thresh" in c and "med" in c]+["all_med"],
+        labels={"value": "median bias"}
+       )
+```
+
+Questions:
+- The values we saw in the CRPS plots are a lot larger than for the median bias plots. Why is this the case? 
```

Comment:
The bias can be lower because it sums both negative and positive errors, which can cancel each other out. With the CRPS the errors are squared.